javascriptv3/example_code/cross-services/transcribe-streaming-app/src/index.js

```diff
@@ -0,0 +1,208 @@
+  }
+  return Buffer.from(buffer);
+};
+
+window.startRecord = async () => {
+  try {
+    console.log("Recording started");
+    var record = document.getElementById("record");
+    var stop = document.getElementById("stopRecord");
+    record.disabled = true;
+    record.style.backgroundColor = "blue";
+    stop.disabled = false;
+
+    // Start the browser microphone.
+    const micStream = new MicrophoneStream();
+    micStream.setStream(
+      await window.navigator.mediaDevices.getUserMedia({
+        video: false,
+        audio: true,
+      })
+    );
+
+    // Acquire the microphone audio stream.
+    const audioStream = async function* () {
+      for await (const chunk of micStream) {
+        yield {
+          AudioEvent: {
+            AudioChunk: pcmEncodeChunk(
+              chunk
+            ) /* pcm Encoding is optional depending on the source. */,
+          },
+        };
+      }
+    };
+
+    const command = new StartStreamTranscriptionCommand({
+      // The language code for the input audio. Valid values are en-GB, en-US, es-US, fr-CA, and fr-FR.
+      LanguageCode: "en-US",
+      // The encoding used for the input audio. The only valid value is pcm.
+      MediaEncoding: "pcm",
+      // The sample rate of the input audio in Hertz.
+      MediaSampleRateHertz: 44100,
+      AudioStream: audioStream(),
+    });
+
+    // Send the speech stream to Amazon Transcribe.
+    const data = await transcribeClient.send(command);
+    console.log("Success", data.TranscriptResultStream);
+    //return data; //For unit tests only.
+    for await (const event of data.TranscriptResultStream) {
+      for (const result of event.TranscriptEvent.Transcript.Results || []) {
+        if (result.IsPartial === false) {
+          const noOfResults = result.Alternatives[0].Items.length;
+          // Print results to browser window.
+          for (let i = 0; i < noOfResults; i++) {
+            console.log(result.Alternatives[0].Items[i].Content);
+            const outPut = result.Alternatives[0].Items[i].Content + " ";
+            const outputDiv = document.getElementById("output");
+            outputDiv.insertAdjacentHTML("beforeend", outPut);
+          }
+        }
+      }
+    }
+    console.log("Success. ", data);
+    client.destroy();
+  } catch (err) {
+    console.log("Error. ", err);
+  }
+};
+
+window.stopRecord = function () {
+  const micStream = new MicrophoneStream();
+  console.log("Recording stopped");
+  var record = document.getElementById("record");
+  var stop = document.getElementById("stopRecord");
+  record.disabled = false;
+  stop.disabled = true;
+  record.style.backgroundColor = "red";
+  micStream.stop();
+};
+
+window.translateText = async () => {
+  try {
+    const outPut = document.getElementById("output").innerHTML;
+    const data = await comprehendClient.send(
+      new DetectDominantLanguageCommand({ Text: outPut })
+    );
+   // return data; //For unit tests only.
```

Comment:
Return
Why the extra comment chars?

Reply:
Replaced with '// Uncomment this for unit tests.'